nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -369,41 +375,40 @@ in
       wants = [ "network-online.target" ];
 
       serviceConfig = {
-        Type = "forking";
-        KillMode = "process";
-        ExecStart = "${wrappedSlurm}/bin/slurmd";
-        PIDFile = "/run/slurmd.pid";
+        Delegate="Yes";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+        # The 23.02.06.01 release assumes slurm*d run as Type = "simple",
+        # with the `-D` flag ensuring the daemons stay in foreground.
+        # Already in 23.11.0.0-rc1 the template .service files
+        # suggest to use Type = "notify", and introduce the `--systemd`
+        # flag instead of `-D`. We'll want to update this line with the release:
+        ExecStart = "${wrappedSlurm}/bin/slurmd -D";
+        KillMode = "process";
         LimitMEMLOCK = "infinity";
-        Delegate="Yes";
+        LimitNOFILE = 131072;
+        LimitSTACK = "infinity";
+        RuntimeDirectoryMode = "0755";
+        RuntimeDirectory = "slurm";
+        Type = "simple";
       };
     };
 
-    systemd.tmpfiles.rules = mkIf cfg.client.enable [
-      "d /var/spool/slurmd 755 root root -"
-    ];
-
     services.openssh.settings.X11Forwarding = mkIf cfg.client.enable (mkDefault true);
 
     systemd.services.slurmctld = mkIf (cfg.server.enable) {
       path = with pkgs; [ wrappedSlurm munge coreutils ]
         ++ lib.optional cfg.enableSrunX11 slurm-spank-x11;
 
       wantedBy = [ "multi-user.target" ];
-      after = [ "network.target" "munged.service" ];
+      after = [ "network-online.target" "munged.service" ];
       requires = [ "munged.service" ];
 
       serviceConfig = {
-        Type = "forking";
-        ExecStart = "${wrappedSlurm}/bin/slurmctld";
-        PIDFile = "/run/slurmctld.pid";
+        Type = "simple";
+        ExecStart = "${wrappedSlurm}/bin/slurmctld -D";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+        LimitNOFILE = 65536;
```

Comment:
Cf. e.g. https://github.com/SchedMD/slurm/blob/slurm-23-02-6-1/etc/slurmctld.service.in